docs/contracts/v4/guides/04-hooks/setup.mdx

```diff
@@ -0,0 +1,223 @@
+---
+title:  Set Up Your Local Environment
+---
+
+Before writing our hook let's first have our local environment properly configured e.g. deploying pool manager, utility routers and test tokens. 
+
+At the end of this guide you’ll have a development environment set up that you can use to build the rest of the examples in the Guides section of the docs, or start your own integration project!
+
+To get you started as quickly as possible for building Uniswap v4 hooks, we have provided the `Quick Start` section below where you can clone a boilerplate and get building. To start from scratch and learn the underlying concepts, jump to the `Start from Scratch` section.
+
+# Quick Start
+
+The Uniswap [v4-template repo](https://github.com/uniswapfoundation/v4-template) provides a basic foundry environment with required imports already pre-loaded for you. You can click on `Use this template` to create a new repository with it.
+
+Or simply clone it and install the dependencies:
+
+```bash
+git clone https://github.com/uniswapfoundation/v4-template.git
+cd v4-template
+# requires foundry
+forge install
+forge test
+```
+
+Then hop to the [Local Node via Anvil](#local-node-via-anvil) to complete your set up and start developing.
+
+# Start from Scratch
+
+In the following sections, we’ll walk through the steps to create the same environment set up as the boilerplate from scratch and learn the underlying concepts.
+
+## Setting up Foundry
+
+The first thing we'll do is set up a new Foundry project. 
+
+If you don't have Foundry installed on your computer already - you can follow the [Foundry Book](https://book.getfoundry.sh/getting-started/installation) for installation.
+
+Once Foundry is setup, we will first initialize a new project:
+
+```bash
+forge init counter-hook
+cd counter-hook
+```
+
+Then we will install the Uniswap `v4-core` and `v4-periphery` contracts as dependencies:
+
+```
+forge install Uniswap/v4-core && forge install Uniswap/v4-periphery
+```
```

Comment:
theres a few code blocks that can be tagged with `bash` or `toml`

Reply:
Sure, just thought it would look better without bash formatting on the forge commands